repair.go

```diff
@@ -20,20 +18,18 @@ import (
 func repairBadIndexVersion(logger log.Logger, dir string) error {
 	// All blocks written by Prometheus 2.1 with a meta.json version of 2 are affected.
 	// We must actually set the index file version to 2 and revert the meta.json version back to 1.
-	subdirs, err := fileutil.ReadDir(dir)
+	dirs, err := blockDirs(dir)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "list block dirs %s", dir)
```

Comment:
I'd prefer `%q` instead of `%s` here and below (especially here, if the dir contains spaces or such).

Should it read "list block dirs %q" -> "list block dirs in %q"?

Reply:
Fair point. Done.